### PR TITLE
Downgrade ts-node, script type fixes

### PR DESCRIPTION
**Pull request by ecraig12345**
Downgrade ts-node to version 7, which has caching (see #12789 discussion). 

Fix some type issues in script files.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12824)

### CODE REVIEW

File: scripts/gulp/plugins/util/getComponentInfo.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import parseDocblock from './parseDocblock';
 import parseType from './parseType';
 import getShorthandInfo from './getShorthandInfo';
 
-const getAvailableBehaviors = (accessibilityProp: ComponentProp): BehaviorInfo[] => {
+const getAvailableBehaviors = (accessibilityProp: ComponentProp | undefined): BehaviorInfo[] | undefined => {
```
Comment (assuncaocharles) [inline scripts/gulp/plugins/util/getComponentInfo.ts:14]:
```suggestion
const getAvailableBehaviors = (accessibilityProp?: ComponentProp): BehaviorInfo[] | undefined => {
```

File: scripts/gulp/plugins/util/getComponentInfo.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import parseDocblock from './parseDocblock';
...
   const docTags = accessibilityProp && accessibilityProp.tags;
```
Comment (assuncaocharles) [inline scripts/gulp/plugins/util/getComponentInfo.ts:15]:
```suggestion
  const docTags = accessibilityProp?.tags;
```

File: scripts/gulp/plugins/util/getComponentInfo.ts [hunk 3]
```diff
@@ -87,7 +87,7 @@ const getComponentInfo = (tsConfigPath: string, filepath: string, ignoredParentI
   const isChild = !isParent;
   const parentDisplayName = isParent ? null : dirname;
   // "Field" for "FormField" since it is accessed as "Form.Field" in the API
-  const subcomponentName = isParent ? null : info.displayName.replace(parentDisplayName, '');
+  const subcomponentName = isParent ? null : info.displayName.replace(parentDisplayName!, '');
```
Comment (assuncaocharles) [inline scripts/gulp/plugins/util/getComponentInfo.ts:90]: Didn't got why the `!` added

File: scripts/gulp/plugins/util/getComponentInfo.ts (hunk 1 above)
Comment (layershifter) [reply scripts/gulp/plugins/util/getComponentInfo.ts:14]: Nope.

```ts
function a(a: string | undefined) { };
function b(a?: string) { };

console.log(b('a'))
// really optional
console.log(b())

console.log(a('a'))
// requires to pass a param
console.log(a())
```

https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=11&pc=17#code/GYVwdgxgLglg9mABAQwBTIFyIM5QE4xgDmiAPouACYCmwh1lAlIgN6IC+A3ALABQokWAkQAjdAH4suAsWZsufRbwgJscADbUAdOrhFUYgOTJDjRnwD0FxHmrJ16gJ6I4AByFh7fFWDWadegaoZko+ftq6+mjGpua8VjbUAI4gMLbYiFBwiK7I2BnIOch4yAC23qoaEYFoZkA

Comment (assuncaocharles) [reply scripts/gulp/plugins/util/getComponentInfo.ts:14]: Alright, makes sense but in this case makes really difference?

Comment (ecraig12345) [reply scripts/gulp/plugins/util/getComponentInfo.ts:14]: In this case the param is always passed, so we'll leave it as `ComponentProp | undefined`.

File: scripts/gulp/plugins/util/getComponentInfo.ts (hunk 3 above)
Comment (ecraig12345) [reply scripts/gulp/plugins/util/getComponentInfo.ts:90]: It's a non-null assertion operator (needed here when `strictNullChecks` is enabled). `parentDisplayName` will never actually be null in this case, but TS is unable to infer that for some reason.

Comment (assuncaocharles) [reply scripts/gulp/plugins/util/getComponentInfo.ts:90]: Alright, makes sense. Thanks :)

File: scripts/gulp/plugins/util/getComponentInfo.ts (hunk 1 above)
Comment (assuncaocharles) [reply scripts/gulp/plugins/util/getComponentInfo.ts:14]: Alright :)

**Pull request merged**